pkgs/development/compilers/sbcl/2.x.nix

```diff
@@ -41,29 +41,13 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ texinfo ];
   buildInputs = lib.optionals coreCompression [ zstd ];
 
-  # There are no patches necessary for the currently enabled versions, but this
-  # code is left in place for the next potential patch.
-  postPatch = ''
-    echo '"${version}.nixos"' > version.lisp-expr
-
-    # SBCL checks whether files are up-to-date in many places..
-    # Unfortunately, same timestamp is not good enough
-    sed -e 's@> x y@>= x y@' -i contrib/sb-aclrepl/repl.lisp
-    #sed -e '/(date)/i((= date 2208988801) 2208988800)' -i contrib/asdf/asdf.lisp
-    sed -i src/cold/slam.lisp -e \
-      '/file-write-date input/a)'
-    sed -i src/cold/slam.lisp -e \
-      '/file-write-date output/i(or (and (= 2208988801 (file-write-date output)) (= 2208988801 (file-write-date input)))'
-    sed -i src/code/target-load.lisp -e \
-      '/date defaulted-fasl/a)'
-    sed -i src/code/target-load.lisp -e \
-      '/date defaulted-source/i(or (and (= 2208988801 (file-write-date defaulted-source-truename)) (= 2208988801 (file-write-date defaulted-fasl-truename)))'
-
-    # Fix the tests
-    sed -e '5,$d' -i contrib/sb-bsd-sockets/tests.lisp
-    sed -e '5,$d' -i contrib/sb-simple-streams/*test*.lisp
-  ''
-  + (if purgeNixReferences
+  postPatch = (lib.optionalString (builtins.elem stdenv.hostPlatform.system [
+    "x86_64-linux"
+    "x86_64-darwin"
+    "aarch64-linux"
+  ]) ''
+    rm tests/foreign-stack-alignment.impure.lisp
```

Comment:
This test fails on x86_64-darwin, but only when run from within a `nix develop ...#sbcl`. Not if I just build sbcl directly on the host. So there's something nixpkgs-specific going on.. My OS is clang 14; I've tried from nixpkgs with clang 11 (default) and 16 and neither worked ( I guess I could try with 14 but I assume at this point that won't make a difference). I'm not sure what else is different.. maybe which zlib is being linked for core compression?

To complicate matters further: this test works fine on aarch64-darwin. The plot thickens!